### ISSUE TITLE
Compiler Crash on empty tuple types

### ISSUE THREAD

**Issue by jppresents** [Microsoft/TypeScript#17925]
**TypeScript Version:**  2.4.1

**Code**

```ts
class What {
    x: [];
    constructor() {
        this.x = [];
    }
}
```

**Expected behavior:**
compiler error for wrong array type declaration.

**Actual behavior:**
Compiler Crash:
Process: Error processing message: Debug Failure.  Error: Debug Failure. 
    at typeToString (evalmachine.<anonymous>:26117:22)
    at reportRelationError (evalmachine.<anonymous>:31446:34)
    at isRelatedTo (evalmachine.<anonymous>:31563:21)
    at checkTypeRelatedTo (evalmachine.<anonymous>:31429:26)
    at checkTypeAssignableTo (evalmachine.<anonymous>:31169:20)
    at checkAssignmentOperator (evalmachine.<anonymous>:37927:25)
    at checkBinaryLikeExpression (evalmachine.<anonymous>:37888:21)
    at checkBinaryExpression (evalmachine.<anonymous>:37766:20)
    at checkExpressionWorker (evalmachine.<anonymous>:38202:28)
    at checkExpression (evalmachine.<anonymous>:38125:42)

**Comment by ikatyang** [comment]
Fixed by #17311. 

(`typescript@rc` / `typescript@next`)

```ts
class What {
    x: [];
    // ^^ [ts] A tuple type element list cannot be empty.   
    constructor() {
        this.x = [];
    }
}
```

**Comment by DanielRosenwasser** [comment]
Thanks for finding the duplicate (#17303) @ikatyang!
